assignments/Session_02/http_server.py

```diff
@@ -0,0 +1,123 @@
+import socket
+import sys
+import os
+import mimetypes
+
+
+def response_ok(body=b"this is a pretty minimal response", mimetype=b"text/plain"):
+    """returns a basic HTTP response"""
+    resp = []
+    resp.append(b"HTTP/1.1 200 OK")
+    resp.append(b"Content-Type: " + mimetype)
+    resp.append(b"")
+    resp.append(body)
+    return b"\r\n".join(resp)
+
+
+def response_method_not_allowed():
+    """returns a 405 Method Not Allowed response"""
+    resp = []
+    resp.append("HTTP/1.1 405 Method Not Allowed")
+    resp.append("")
+    return "\r\n".join(resp).encode('utf8')
+
+
+def response_not_found():
+    """returns a 404 Not Found response"""
+    return b"HTTP/1.1 404 Not Found"
```

Comment:
This is not a complete HTTP response.  You are missing the blank line that separates the header section  from the body of the response.  Even though there is no body to this type of response, it's still required to have that blank line.